libraries/standard/http/src/http_client.c

```diff
@@ -40,21 +139,149 @@ HTTPStatus_t HTTPClient_InitializeRequestHeaders( HTTPRequestHeaders_t * pReques
 /*-----------------------------------------------------------*/
 
 HTTPStatus_t HTTPClient_AddHeader( HTTPRequestHeaders_t * pRequestHeaders,
-                                   const char * pName,
-                                   size_t nameLen,
+                                   const char * pField,
+                                   size_t fieldLen,
                                    const char * pValue,
                                    size_t valueLen )
 {
-    return HTTP_NOT_SUPPORTED;
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+
+    /* Check for NULL parameters. */
+    if( pRequestHeaders == NULL )
+    {
+        IotLogError( "Parameter check failed: pRequestHeaders interface is NULL." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( pRequestHeaders->pBuffer == NULL )
+    {
+        IotLogError( "Parameter check failed: pRequestHeaders->pBuffer is NULL." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( ( pField == NULL ) )
+    {
+        IotLogError( "Parameter check failed: pField is NULL." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( ( pValue == NULL ) )
+    {
+        IotLogError( "Parameter check failed: pValue is NULL." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( fieldLen == 0u )
+    {
+        IotLogError( "Parameter check failed: fieldLen must be greater than 0." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( valueLen == 0u )
+    {
+        IotLogError( "Parameter check failed: valueLen must be greater than 0." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else
+    {
+        /* Empty else MISRA 15.7 */
+    }
+
+    if( returnStatus == HTTP_SUCCESS )
+    {
+        returnStatus = _addHeader( pRequestHeaders,
+                                   pField, fieldLen, pValue, valueLen );
+    }
+
+    return returnStatus;
 }
 
 /*-----------------------------------------------------------*/
 
 HTTPStatus_t HTTPClient_AddRangeHeader( HTTPRequestHeaders_t * pRequestHeaders,
-                                        int32_t rangeStart,
+                                        int32_t rangeStartOrlastNbytes,
                                         int32_t rangeEnd )
 {
-    return HTTP_NOT_SUPPORTED;
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+    char rangeValueBuffer[ MAX_RANGE_REQUEST_VALUE_LEN ] = { 0 };
```

Comment:
+1 for the null terminator that snprintf adds?

Reply:
Can we add a test that writes the maximum integers for rangeStart and rangeEnd?

Reply:
Good point. I do have a [stress test](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/f67e9c46ed2f300b36b702d3186f6f161d392a92/libraries/standard/http/test/test-HTTPClient_AddRangeHeader.c#L211-L229) with the `INT32_MAX` values for `rangeStart` and `rangeEnd` which runs fine without the NULL (surprisingly). Have added the extra byte space.